### ISSUE TITLE
ASB 2018.02

### ISSUE THREAD

**Issue by MSe1969** [cm12-amami/discussion#20]
February ASB is out, yet w/o AOSP links...

**Comment by MSe1969** [comment]
***2018-02-01 SECTION in Android-Bulletin***
**Media Framework**
- n/a  CVE-2017-13228   A-69478425  - not for LP
- n/a CVE-2017-13231   A-67962232  - not for LP
- [x] CVE-2017-13232    A-68953950  - applied and pushed  
- [x] CVE-2017-13230    A-65483665  - applied and pushed
- [x] CVE-2017-13233    A-62851602  - applied and pushed
- [x] CVE-2017-13234    A-68159767  - applied and pushed

**System**
- n/a CVE-2017-13236   A-68217699 [2]  - not for LP


***2018-02-05 SECTION in Android-Bulletin***

**Kernel**
- [x] CVE-2017-15265  A-67900971  - fixed & pushed
- n/a CVE-2015-9016 A-63083046 - 3 files do not exist and the patches in the other 2 refer to them
- [x] CVE-2017-17770  A-65853158* - found, applied and pushed

**Qualcomm components**
- [x] CVE-2017-15817   A-68992394   -  fixed & pushed
- [x] CVE-2017-17760  A-68992416  - seems applicable, but code differences
- n/a CVE-2017-11041  A-35269676*  -  found (hardware/qcom/media), but n/a for cm12.1 msm8974
- n/a CVE-2017-17767  A-64750179*  -  not found, not publicly available
- n/a: CVE-2017-17765  A-68992445 - file wma_utils.c does not exist
- n/a: CVE-2017-17762  A-68992439 - file wma_utils.c does not exist
- n/a: CVE-2017-14884  A-68992429 - function 'WLANQCMBR_McProcessMsg' not present
- n/a: CVE-2017-15829  A-68992397 - functions to be patched not present in kgsl.c
- n/a: CVE-2017-15820  A-68992396 - functions to be patched not present in kgsl.c
- n/a: CVE-2017-17764  A-68992443 - file wma_features.c does not exist
- n/a: CVE-2017-17761  A-68992434 - file wma_scan_roam.c does not exist

***PIXEL-Bulletin***

**Framework**
- n/a CVE-2017-13239   A-66244132* - not for LP
- n/a CVE-2017-13240   A-68694819  - not for LP

**Media-Framework**
- [x] CVE-2017-13241   A-69065651  - fixed & pushed
- [x] CVE-2017-13229   A-68160703  - fixed & pushed
- [x] CVE-2017-13235   A-68342866  - fixed & pushed

**System**
- n/a CVE-2017-13242   A-62672248 [2] - not for LP
- n/a CVE-2017-13243   A-38258991*    - not found, not publicly available

**Kernel**
- n/a CVE-2017-13244   A-62678986*  - not found, not publicly available (maybe diff. kernel)
- [x] CVE-2017-13245   A-64315347*  - found, applied and pushed
- n/a CVE-2017-1000405 A-69934280   - functions 'touch_pmd'/'touch_pud' don't exist 
- [x] CVE-2017-13246   A-36279469*  - found, applied and pushed

**Qualcomm components**
- n/a CVE-2017-15859   A-65468985   - fn 'wlan_hdd_cfg80211_txpower_scale' doesn't exist
- [x] CVE-2017-17769   A-65172622*  - found, applied and pushed
- n/a CVE-2017-9723    A-68992479   - files do not exist
- n/a CVE-2017-14881   A-68992478   - code section to be patched doesn't exist
- n/a CVE-2017-14877   A-68992473   - function 'ipa_query_rt_index' does not exist
- n/a CVE-2017-15826    A-68992471  -  file mdss_totator.c does not exist
- n/a CVE-2017-14876    A-68992468  -  func 'msm_ispif_config_stereo' doesn't exist
- n/a CVE-2017-14892    A-68992455  -  func 'q6asm_set_shared_circ_buff' doesn't exist
- n/a CVE-2017-17766    A-68992448  -  file wma_utils.c doesn't exist
- n/a CVE-2017-15823    A-68992447  -  file wma_features.c doesn't exist
- n/a CVE-2017-15852    A-36730614* -  not found, not publicly available (maybe diff. kernel)
- n/a CVE-2017-15846    A-67713103  -  file */ais/msm.c: subdir /ais/ doesn't exist
- n/a CVE-2017-14883    A-68992426  -  file wma_features.c doesn't exist
- n/a CVE-2017-11043    A-68992421  -  none of the to-be-patched files exists
- n/a CVE-2017-14875    A-68992465  -  fn 'msm_isp_ab_ib_update_lpm_mode' doesn't exist
- n/a CVE-2017-14891    A-68992453  -  fn '_gpuobj_map_useraddr' doen't exist
- n/a CVE-2017-17771    A-38196031  -  'qbuf_buf->num_planes' does not exist
- n/a CVE-2017-11087    A-34735194* - not found, not publicly available (maybe diff. kernel)


NOTE: 
- Unchecked checkbox means "to do"! Patches, which are not applicable are marked accordingly
- The "missing AOSP links" patches will be looked after, once the new AOSP tag os out

**Comment by MSe1969** [comment]
@hahnjo @Micha-Btz - I've gone through the 2018-02-05 sections already and listed all above:
CVE-2015-9016, CVE-2017-17760, CVE-2017-17769, CVE-2017-17771 need a closer look to perform a backport, if one of you has time, would be appreciated
Besides, let's wait until the AOSP links are published

**Comment by MSe1969** [comment]
Applied Android Bulletin, 2018-02-01 section now

**Comment by Micha-Btz** [comment]
I will take a look this weekend.

**Comment by MSe1969** [comment]
Okay, almost done - the following items are still a to-do from above list:
*Part 1 - backports*
CVE-2015-9016 - to backport or search/apply further commits (code differences)
CVE-2017-17760 - to backport or search/apply further commits (code differences)

*Part 2 - remove a driver*
CVE-2017-9723
we should do as **very last** action, as it removes a driver, kernel should run stable before, so we can isolate any issue to this driver, which would then in our case still be needed
Proposal: Ship first, and then do it afterwards and push to stable branch (and ship with March ASB)

**Comment by Micha-Btz** [comment]
I have pushed the fix for CVE-2017-17760 please test the wlan's in your environment.

**Comment by Micha-Btz** [comment]
to CVE-2015-9016 a lots of file are missing in our code
~/handy/build/cm-12.1/kernel/sony/msm8974/block] [:56fe32c8d76|AM 1/1|] ✔ 
```
 $ ll blk.h blk-mq.c blk-mq-tag.h blk-mq-tag.c blk-flush.c
ls: Zugriff auf 'blk-mq.c' nicht möglich: Datei oder Verzeichnis nicht gefunden
ls: Zugriff auf 'blk-mq-tag.h' nicht möglich: Datei oder Verzeichnis nicht gefunden
ls: Zugriff auf 'blk-mq-tag.c' nicht möglich: Datei oder Verzeichnis nicht gefunden
-rw-r--r-- 1 mdomann mdomann  14K Feb 10 20:57 blk-flush.c
-rw-r--r-- 1 mdomann mdomann 7,3K Feb 10 20:57 blk.h
```

In blk-flush.c is blk-mq-tag.h to include which also not exist. Also the function which is patched doesn't exist. I will leave this out as n/a.

CVE-2017-9723 the files for this aren't there too. So I think we are ready to go.

**Comment by Micha-Btz** [comment]
Will you check for  CVE-2017-9723?

**Comment by MSe1969** [comment]
Thanks, I fully agree. There is only one last item:
CVE-2017-0846 A-64934810[2] from the January Pixel bulletin - it did not work in cm-13.0, so we skipped it there and I also skipped in our cm-12.1 and added it into our open patches issue.
In cm-11.0 at LineageOS, the guys have developed and merged a backport, I'll try those commits as well and test tomorrow. 
https://review.lineageos.org/#/c/202488/
https://review.lineageos.org/#/c/205557/
If all works, I'll ship then.
